devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1365,6 +1866,579 @@ def deleteCode(codeID) {
 	], 4200)
 }
 
+/**
+ * API end-point for updating the 'lockCodesData' attribute with the user info
+ *
+ * @param userInfo: Map consisting values for code id, phone number, access type, and last usage
+ */
+def setUserInfo(userInfo) {
+	log.trace "[DTH] Executing 'setUserInfo()' by ${device.displayName} with userInfo := ${userInfo}"
+	def codeID = userInfo.codeID
+	if (codeID) {
+		codeID = codeID.toString()
+		def lockCodesData = loadlockCodesData(codeID)
+		lockCodesData[codeID].lastUpdated = userInfo.lastUpdated
+		lockCodesData[codeID].userInfo.phoneNum = userInfo.phoneNum ?: ""
+		def accessType = userInfo.accessType
+		if (accessType) {
+			if (accessType == "always") {
+				lockCodesData[codeID].scheduleInfo = [:]
+				lockCodesData[codeID].scheduleInfo.scheduleData = [:]
+			}
+			lockCodesData[codeID].scheduleInfo.scheduleType = accessType
+		}
+		sendEvent(lockCodesDataEvent(lockCodesData))
+	}
+}
+
+/**
+ * Starts the scanning of schedule for the first code present in the 'lockCodes' map
+ */
+def startScanningSchedule() {
+	log.trace "[DTH] Executing 'startScanningSchedule()' by ${device.displayName}"
+	def codesToScanSchedules = state.codesToScanSchedules
+	def codeID = -1
+	if (codesToScanSchedules && codesToScanSchedules.size() > 0) {
+		// Schedule scanning did not complete the last time. Scan the remaining schedules.
+		codeID = codesToScanSchedules[0]
+	} else {
+		def lockCodes = loadLockCodes()
+		if (lockCodes && lockCodes.size() > 0) {
+			def keySet = lockCodes.keySet()
+			codesToScanSchedules = []
```

Comment:
In the creation/assignment above, you could just do

`def codesToScanSchedules = state.codesToScanSchedules ?: []`

and avoid this line.